### ISSUE TITLE
Back-end check for "required".

### ISSUE THREAD

**Issue by krazyjakee** [roundhouse/FormBuilder-Craft-CMS#32]
A back-end check needs to exist for required fields as older browsers do not support the html5 tag and/or have javascript disabled.

**Comment by owldesign** [comment]
Good point, on my list

**Comment by nevsie** [comment]
Has this been actioned? JUst asking as i see no notes to it?

**Comment by owldesign** [comment]
I have not added this yet, have been busy with other work can't find time to update yet.

**Comment by cole007** [comment]
+1 for this feature please

**Comment by krazyjakee** [comment]
Not sure if this is controversial but I have a fork where considerable changes have been made including back-end validation. Of course, happy for any logic to be taken and used in roundhouse version but the changes are too considerable to warrant a pull-request.

**Comment by owldesign** [comment]
I have added server side validation to the available fieldtypes. 

At the moment the fields are not retained if validation fails, so users will have to fill out the form again. I'll be digging further when I get more time, but at least for now the forms can be validated on older (and safari) browsers that don't support `required` field.

You'll need to add this snipped to get the error messages (sampleForm.html) has been updated as well.

`{% if errors is defined and errors | length %}
      <ul class="errors">
      {% for error in errors %}
       <li class="error-message">{{ error }}</li>
      {% endfor %}
      </ul>
    {% endif %}`
